src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -119,30 +127,35 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
     IntermediateOutputPath:
     This is the full intermediate Output Path, and is derived from BaseIntermediateOutputPath, if none specified
-    (eg. obj\debug). If this property is overridden, then setting BaseIntermediateOutputPath has no effect.
+    (eg. obj\Debug). If this property is overridden, then setting BaseIntermediateOutputPath has no effect.
+
+    Ensure any and all path property has a trailing slash, so it can be concatenated.
     -->
 
   <PropertyGroup>
-    <!-- Ensure any OutputPath has a trailing slash, so it can be concatenated -->
-    <OutputPath Condition="'$(OutputPath)' != '' and !HasTrailingSlash('$(OutputPath)')">$(OutputPath)\</OutputPath>
+    <BaseOutputPath Condition=" '$(BaseOutputPath)' == '' ">bin\</BaseOutputPath>
+    <BaseOutputPath Condition="!HasTrailingSlash('$(BaseOutputPath)')">$(BaseOutputPath)\</BaseOutputPath>
+
     <AssemblyName Condition=" '$(AssemblyName)'=='' ">$(MSBuildProjectName)</AssemblyName>
     <!--
         Be careful not to give OutputPath a default value in the case of an invalid Configuration/Platform.
         We use OutputPath specifically to check for invalid configurations/platforms.
         -->
-    <OutputPath Condition=" '$(Platform)'=='' and '$(Configuration)'=='' and '$(OutputPath)'=='' ">bin\Debug\</OutputPath>
+    <OutputPath Condition=" '$(Platform)'=='' and '$(Configuration)'=='' and '$(OutputPath)'=='' ">$(BaseOutputPath)</OutputPath>
     <_OriginalConfiguration>$(Configuration)</_OriginalConfiguration>
     <_OriginalPlatform>$(Platform)</_OriginalPlatform>
     <Configuration Condition=" '$(Configuration)'=='' ">Debug</Configuration>
-    <ConfigurationName Condition=" '$(ConfigurationName)' == '' ">$(Configuration)</ConfigurationName>
+    <_Configuration Condition=" '$(_Configuration)' == '' ">$(Configuration)</_Configuration>
```

Comment:
What's the motivation for this change? Is it separable from the other part?

Reply:
Are `ConfigurationName` and `PlatformName` is being used anywhere besides the Common targets?

I have looked in all the targets that shipped with Dev10 until now and I didn't find any. In the refactored version, I removed all these workarounds and used them directly.

Reply:
They are "public"--not prefixed with a `_`--and can be accessed by arbitrary build logic in user projects. As such, removing them is a breaking change.

Reply:
OK. I'll revert them. So, is it meant to be an alias for the original properties?!

Reply:
No idea what the intent was, unfortunately. But we have enough experience to be very scared of making changes.